packages/react-ui-components/src/components/SchemaForm/Widgets/SelectWidget/SelectWidget.tsx

```diff
@@ -1,47 +1,75 @@
-import React, { useMemo, useState } from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import { WidgetProps } from '@rjsf/core';
 import { KvMultiSelectDropdown, KvSingleSelectDropdown } from '../../../stencil-generated';
 import { EValidationState } from '@kelvininc/ui-components';
-import { buildSelectedOptions, getSelectedOptions, processValue } from './utils';
+import { buildDropdownOptions, buildSelectedOptions, getSelectedOptions, getWidgetName, processValue, searchDropdownOptions } from './utils';
 
-const SelectWidget = ({ schema, id, options, label, required, disabled, readonly, value, multiple, onChange, placeholder, rawErrors = [] }: WidgetProps) => {
+const SelectWidget = ({ schema, id, options, label, required, disabled, readonly, value, multiple, onChange, placeholder, rawErrors = [], formContext = {} }: WidgetProps) => {
 	const { enumOptions, enumDisabled } = options;
+	const { displayValue, searchable, selectionClearable } = formContext[getWidgetName(id)] ?? {};
+
+	const defaultDropdownOptions = useMemo(() => buildDropdownOptions(enumOptions, enumDisabled), [enumOptions, enumDisabled]);
+	const [searchTerm, setSearchTerm] = useState<string | null>(null);
+	const [searchedOptions, setSearchedOptions] = useState(defaultDropdownOptions);
+	const [stateValue, setStateValue] = useState(processValue(schema, value));
+
 	const emptyValue = useMemo(() => (multiple ? [] : undefined), [multiple]);
+
+	const onSearchChange = useCallback(({ detail: searchedLabel }: CustomEvent<string>) => setSearchTerm(searchedLabel), []);
+	const onChangeOptionSelected = useCallback(({ detail: selectedOption }: CustomEvent<string>) => {
+		onChangeValue(selectedOption);
+	}, []);
+	const onChangeOptionsSelected = useCallback(({ detail: selectedOptionsMap }: CustomEvent<{ [key: string]: boolean }>) => {
+		const selectedOptions = getSelectedOptions(selectedOptionsMap);
+		onChangeValue(selectedOptions);
+	}, []);
+	const onSelectionCleared = useCallback(() => {
+		onChangeValue([]);
+	}, [onChange]);
+	const onChangeValue = useCallback(newValue => {
+		const processedValue = processValue(schema, newValue);
+		onChange(processedValue);
+		setStateValue(processedValue);
+	}, []);
+
 	const args = {
 		id,
 		label: label || schema.title,
 		placeholder: placeholder ? placeholder : undefined,
 		required,
 		disabled: disabled || readonly,
 		errorState: rawErrors.length > 0 ? EValidationState.Invalid : EValidationState.Valid,
-		displayValue: typeof value === 'undefined' ? emptyValue : value,
-		options: Array.isArray(enumOptions)
-			? enumOptions.reduce((acc, { value, label }) => {
-					const disabled = Array.isArray(enumDisabled) && enumDisabled.indexOf(value) != -1;
-					acc[value] = { value, label, disabled };
-					return acc;
-			  }, {})
-			: []
+		displayValue: typeof value === 'undefined' ? emptyValue : displayValue?.(value, enumOptions),
```

Comment:
Please add a story that uses this DISPLAY VALUE.

Reply:
I'll do 👍